dwave/optimization/src/nodes/flow.cpp

```diff
@@ -42,6 +42,113 @@ std::span<const ssize_t> same_shape(const Array* node_ptr,
     return node_ptr->shape();
 }
 
+/// ExtractNode
+
+ExtractNode::ExtractNode(ArrayNode* condition_ptr, ArrayNode* arr_ptr)
+        : ArrayOutputMixin({-1}), condition_ptr_(condition_ptr), arr_ptr_(arr_ptr) {
+    if (condition_ptr_->sizeinfo() != arr_ptr_->sizeinfo()) {
```

Comment:
I think you want `same_shape(condition_ptr_, arr_ptr_)`, which does the recursive check.

Reply:
`same_shape()` actually checks only the non-dynamic part of the shape. And what we care about here is same size, as different shapes are explicitly allowed. But is there some other check I should be doing?

Reply:
You're right, the suggestion was wrong.

BUt I think this check can fail in some cases where it should succeed if we don't do something like
```c++
if (condition_ptr_->sizeinfo().substitute(100) != arr_ptr_->sizeinfo().substitute(100)) {
```

I thought we had a utility function but I cannot find it :dizzy_face: